File: ios/RuneChess/Game/RuneType.swift
import SpriteKit

enum RuneType: CaseIterable {
    case red
    case blue
    case green
    case yellow
    case purple
    case white

    var color: SKColor {
        switch self {
        case .red:
            return SKColor(red: 0.79, green: 0.29, blue: 0.29, alpha: 1.0)
        case .blue:
            return SKColor(red: 0.29, green: 0.49, blue: 0.82, alpha: 1.0)
        case .green:
            return SKColor(red: 0.33, green: 0.63, blue: 0.42, alpha: 1.0)
        case .yellow:
            return SKColor(red: 0.87, green: 0.75, blue: 0.31, alpha: 1.0)
        case .purple:
            return SKColor(red: 0.53, green: 0.38, blue: 0.74, alpha: 1.0)
        case .white:
            return SKColor(red: 0.91, green: 0.89, blue: 0.82, alpha: 1.0)
        }
    }
}

File: Packages/com.runechess.core/Runtime/RuneType.cs
using System;
using System.Collections.Generic;

namespace RuneChess.Core
{
    public enum RuneType
    {
        Red,
        Blue,
        Green,
        Yellow,
        Purple,
        White
    }

    public static class RuneTypes
    {
        public const string RedId = "red";
        public const string BlueId = "blue";
        public const string GreenId = "green";
        public const string YellowId = "yellow";
        public const string PurpleId = "purple";
        public const string WhiteId = "white";

        public static IReadOnlyList<RuneType> All { get; } = Array.AsReadOnly(new[]
        {
            RuneType.Red,
            RuneType.Blue,
            RuneType.Green,
            RuneType.Yellow,
            RuneType.Purple,
            RuneType.White
        });

        public static string GetId(RuneType rune)
        {
            return rune switch
            {
                RuneType.Red => RedId,
                RuneType.Blue => BlueId,
                RuneType.Green => GreenId,
                RuneType.Yellow => YellowId,
                RuneType.Purple => PurpleId,
                RuneType.White => WhiteId,
                _ => throw new ArgumentOutOfRangeException(nameof(rune), rune, "Unknown rune type.")
            };
        }

        public static RuneType ParseId(string id)
        {
            if (TryParseId(id, out var rune))
            {
                return rune;
            }

            throw new ArgumentException($"Unknown rune id '{id}'.", nameof(id));
        }

        public static bool TryParseId(string? id, out RuneType rune)
        {
            if (string.IsNullOrWhiteSpace(id))
            {
                rune = default;
                return false;
            }

            var normalizedId = id.Trim();
            if (normalizedId.Equals(RedId, StringComparison.OrdinalIgnoreCase))
            {
                rune = RuneType.Red;
                return true;
            }

            if (normalizedId.Equals(BlueId, StringComparison.OrdinalIgnoreCase))
            {
                rune = RuneType.Blue;
                return true;
            }

            if (normalizedId.Equals(GreenId, StringComparison.OrdinalIgnoreCase))
            {
                rune = RuneType.Green;
                return true;
            }

            if (normalizedId.Equals(YellowId, StringComparison.OrdinalIgnoreCase))
            {
                rune = RuneType.Yellow;
                return true;
            }

            if (normalizedId.Equals(PurpleId, StringComparison.OrdinalIgnoreCase))
            {
                rune = RuneType.Purple;
                return true;
            }

            if (normalizedId.Equals(WhiteId, StringComparison.OrdinalIgnoreCase))
            {
                rune = RuneType.White;
                return true;
            }

            rune = default;
            return false;
        }
    }
}
